src/cdomains/intDomain.ml

```diff
@@ -13,18 +13,23 @@ sig
   include Lattice.S
   val to_int: t -> int64 option
   val of_int: int64 -> t
+  val of_int_ikind: Cil.ikind -> int64 -> t
```

Comment:
Do you mean `cast_to ik @@ of_int`? Or is this `%` some sort of special syntax ? I've only seen it  for extension nodes so far.

It is the same, yes! I think it is advantageous though to have that extra function available so when writing new code we actually make use of this. In the long run, I think we should aim to remove the normal `of_int` altogether: Even if a domain does not keep track of ranges in a meaningful way, everything that comes from CIL should have an associated `ikind`, and using the `of_int_ikind` forces us to think about this.

Reply:
Ah, `%` is from Batteries and that is not opened in the `intDomain.ml`, that's why it wasn't available.

Reply:
Yes, `%` is just function composition, `[%...]` are extension nodes in ppx.

I see your point, still somehow feels messy. It ties the injection to Cil, and some places where we abstract priorities etc. there's no `ikind`. What do people choose then? `ILongLong`?
Maybe add an optional `ikind` argument to `of_int` instead - that way each module can act according to its range.
`Interval32` for example abstracts everything bigger than `int` as `Int32.(min_int, max_int)`.

Reply:
Together with the other introduced `*_ikind` functions, this seems like the cross-product of these injecting functions and `cast_to`. Is any of them necessary?

Reply:
Yes!
Especially for these `starting_ikind` and `ending_ikind`, it does not hold that it is the same as `starting` composed with `cast_to`.

Reply:
Only for those two due to overflow. For the `of_*` it should be the same (and only needed for this PR).
Also, I'm still for changing them to `val starting : ?ikind:Cil.ikind -> int64 -> t` etc. since the definitions should be the same (currently only not the case for `Interval32` and `DefExc`).

Reply:
I removed `starting_ikind`, `ending_ikind`, `of_bool_ikind`, and `of_int_ikind` now.